sound/soc/sof/loader.c

```diff
@@ -289,22 +289,12 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	/* boot the firmware on the DSP */
 	ret = snd_sof_dsp_run(sdev);
 	if (ret < 0) {
-		dev_err(sdev->dev, "error: failed to reset DSP\n");
-		return ret;
-	}
-
-	/* now wait for the DSP to boot */
-	ret = wait_event_timeout(sdev->boot_wait, sdev->boot_complete,
-				 msecs_to_jiffies(sdev->boot_timeout));
-	if (ret == 0) {
-		dev_err(sdev->dev, "error: firmware boot timeout\n");
+		dev_err(sdev->dev, "error: failed to boot firmware\n");
 		snd_sof_dsp_dbg_dump(sdev, SOF_DBG_REGS | SOF_DBG_MBOX |
-			SOF_DBG_TEXT | SOF_DBG_PCI);
-		return -EIO;
+				     SOF_DBG_TEXT | SOF_DBG_PCI);
+		return ret;
 	}
 
-	dev_info(sdev->dev, "firmware boot complete\n");
-
 	return 0;
```

Comment:
An alternate solution was to add a .pre_load and .post_load callback instead of changing everything just to add clock gating off/on in specific location. this is a bit too much cut/paste for my taste.

Reply:
sure that makes sense. will fix it.

Reply:
From the code context, using pre/post_fw_run is more suitable here.

Reply:
updated now!